[SPARK-8111] [SparkR] SparkR shell should display Spark logo and version banner on startup.

aloknsingh opened the pull request:
spark version is taken from the environment variable SPARK_VERSION

aloknsingh committed:
fix the jira SPARK-8111 to add the spark version and logo. Currently spark version is taken from the environment variable SPARK_VERSION

AmplabJenkins commented:
Can one of the admins verify this patch?

andrewor14 commented:
nice! @shivar

andrewor14 commented:
ok to test. @aloknsingh Can you post a screenshot of what the logo looks like?

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35501 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35501/consoleFull) for   PR 6944 at commit [`acd5b85`](https://github.com/apache/spark/commit/acd5b85fdbaf26ef19b0abca28eace58b7d350b7).

SparkQA commented:
[Test build #35501 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35501/console) for   PR 6944 at commit [`acd5b85`](https://github.com/apache/spark/commit/acd5b85fdbaf26ef19b0abca28eace58b7d350b7).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

shivaram commented on R/pkg/inst/profile/shell.R:
```diff
@@ -29,5 +30,18 @@
   sqlContext <- SparkR::sparkRSQL.init(sc)
   assign("sqlContext", sqlContext, envir=.GlobalEnv)
   cat("\n Welcome to SparkR!")
```
We can remove `SparkR` from here and just make this `Welcome to`. That way the entire sentence will read `Welcome to Spark`

shivaram commented on R/pkg/inst/profile/shell.R:
```diff
@@ -17,6 +17,7 @@
 
 .First <- function() {
   home <- Sys.getenv("SPARK_HOME")
+  sparkVer <- Sys.getenv("SPARK_VERSION")
```
I'm not sure when this environment variable is set. We can instead call the java method `sc.version` after creating the spark context. To do that you can do something like 
```
sparkVer <- SparkR:::callJMethod(sc, "version")
```

shivaram commented:
Thanks @aloknsingh for the change. I had some minor comments inline. Also to reply to your question from the JIRA I'd say we can just keep the ASCII art as `Spark` right now.

Commit:
[SPARK-8111][SparkR] As per suggestion, 1) using the version from sparkContext rather than the Sys.env.
2) change "Welcome to SparkR!" to "Welcome to" followed by Spark logo and version

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35564 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35564/consoleFull) for   PR 6944 at commit [`ed607bd`](https://github.com/apache/spark/commit/ed607bd9bb8398ea8fdf1206ecfb5bf79dd6b7a5).

aloknsingh commented:
@sivaram: I modified the code as per your suggestions.

@andrewor14: Here is the screenshot.
![spark-8111_sparklogo](https://cloud.githubusercontent.com/assets/12959246/8312394/1266fa5a-1991-11e5-99cb-62062a328ed7.png)

SparkQA commented:
[Test build #35564 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35564/console) for   PR 6944 at commit [`ed607bd`](https://github.com/apache/spark/commit/ed607bd9bb8398ea8fdf1206ecfb5bf79dd6b7a5).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

aloknsingh commented:
Jenkins. Please retest.

shivaram commented:
@aloknsingh Actually the tests passed if you look at the Spark QA message (above the jenkins one) -- the jenkins plugin has an issue where it reports failures for some builds.

andrewor14 commented:
retest this please

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #35565 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35565/consoleFull) for   PR 6944 at commit [`ed607bd`](https://github.com/apache/spark/commit/ed607bd9bb8398ea8fdf1206ecfb5bf79dd6b7a5).

shivaram commented:
LGTM. Will merge once tests pass

andrewor14 commented:
Hm, this should have just run the R tests.

shivaram commented:
Yeah I have no idea why that changed -- @JoshRosen did the limited unit test running thing change again ?

JoshRosen commented:
It looks like the pull request builder somehow thinks that the `sql` module changed.

shivaram commented:
hmm maybe it compared this PR to a different cached git commit ? I've seen that produce warnings for new classes added etc. before

JoshRosen commented:
There might have been a race if a PR that changed SQL was committed between the time that the PRB was triggered with a merge commit and the time that we fetched origin/master to do a comparison for changed files.  We can probably fix this by using a different base / SHA to compute the diff.

SparkQA commented:
[Test build #35565 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/35565/console) for   PR 6944 at commit [`ed607bd`](https://github.com/apache/spark/commit/ed607bd9bb8398ea8fdf1206ecfb5bf79dd6b7a5).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.
